WordPress/gutenberg
# Global Styles: Add support for caption elements

## What?
This makes it possible to style captions in theme.json by adding this into your theme.json file:

```
{
	"styles": {
		"elements": {
			"caption": {
				"color": {
					"background": "red",
					"text": "yellow"
				}
			}
		}
	}
}
```

## Why?
So that themes can independently control the display of their captions. See https://github.com/WordPress/gutenberg/issues/40205

## How?
We create a new element called caption and then add the `wp-element-caption` class to all elements that we want to use the caption styles.

The problem with this approach is that captions are also targeted by the default styles in the block themselves. This is not a problem as the CSS global styles outputs is stronger, except for the case of captions inside gallery blocks.

## Testing Instructions
- Add the snippet above to your theme
- Add an audio/image/table/embed/video/gallery block to the page
- Confirm that the captions have the styles added above.

## Screenshots or screencast 
<img width="1415" alt="Screenshot 2022-06-16 at 14 00 22" src="https://user-images.githubusercontent.com/275961/174075248-3f1e0428-fe71-4290-b7ab-5c97553f42ae.png">
